### PR TITLE
Add new mission for using both host versions

### PR DESCRIPTION
**Pull request by sisuresh**
Related to https://github.com/stellar/supercluster/issues/105.

This new mission applies Soroban transactions across protocol versions.

### CODE REVIEW

File: src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    context.Execute
+        [ coreSet; afterUpgradeCoreSet ]
+        None
+        (fun (formation: StellarFormation) ->
+            formation.WaitUntilSynced [ coreSet ]
+
+            let supportedProtocol = (formation.NetworkCfg.GetPeer coreSet 0).GetSupportedProtocolVersion()
+
+            // We want to test against a protocol boundary that has soroban enabled on both sides,
```
Comment (sisuresh) [inline src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs:50]:
```suggestion
            // We want to test against a protocol boundary that has Soroban enabled on both sides,
```

**Review by marta-lokhova — APPROVED** [review]
LGTM!

File: src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                formation.RunLoadgen coreSet context.GenerateAccountCreationLoad
+                formation.UpgradeSorobanLedgerLimitsWithMultiplier [ coreSet ] 1000
+                formation.UpgradeSorobanTxLimitsWithMultiplier [ coreSet ] 100
+                formation.RunLoadgen coreSet context.SetupSorobanInvoke
+                formation.RunLoadgen coreSet context.GenerateSorobanInvokeLoad
+
+                formation.UpgradeProtocol [ coreSet ] (supportedProtocol)
+
+                formation.RunLoadgen coreSet context.GenerateAccountCreationLoad
+                formation.RunLoadgen coreSet context.SetupSorobanInvoke
```
Comment (marta-lokhova) [inline src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs:66]: Actually, is this a no-op? I'd imagine you want to create some protocol 21 Wasms, right?

File: src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+                formation.RunLoadgen coreSet context.GenerateAccountCreationLoad
+                formation.UpgradeSorobanLedgerLimitsWithMultiplier [ coreSet ] 1000
+                formation.UpgradeSorobanTxLimitsWithMultiplier [ coreSet ] 100
+                formation.RunLoadgen coreSet context.SetupSorobanInvoke
+                formation.RunLoadgen coreSet context.GenerateSorobanInvokeLoad
+
+                formation.UpgradeProtocol [ coreSet ] (supportedProtocol)
+
+                formation.RunLoadgen coreSet context.GenerateAccountCreationLoad
```
Comment (marta-lokhova) [inline src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs:65]: This is a no-op. If you want to create new accounts, you'd need to use the `offset` parameter, as account keys are derived from account IDs (between 0+offset and numAccounts+offset)

File: src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs (hunk 2 above)
Comment (sisuresh) [reply src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs:66]: Are you saying the second call to `SetupSorobanInvoke` does nothing?

Comment (sisuresh) [reply src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs:66]: Oh do I need to use the offset here?

File: src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs (hunk 3 above)
Comment (sisuresh) [reply src/FSLibrary/MissionSorobanCatchupWithPrevAndCurr.fs]: This line wasn't necessary so I removed it (but good to know about the offset!)

**Pull request merged**